apps/aesophia_http/src/aesophia_cors_middleware.erl

```diff
@@ -0,0 +1,75 @@
+%%%=============================================================================
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Cross-Origin Resource Sharing (CORS) middleware for Cowboy
+%%% @end
+%%%=============================================================================
+-module(aesophia_cors_middleware).
+
+-behaviour(cowboy_middleware).
+
+%% Behavior API
+-export([execute/2]).
+
+-define(ACCESS_CONTROL_ALLOWED_DOMAINS   , [<<"*">>]).
+-define(ACCESS_CONTROL_ALLOW_METHODS     , <<"DELETE, GET, HEAD, OPTIONS, PATCH, POST, PUT">>).
+-define(ACCESS_CONTROL_MAX_AGE           , 30 * 60). %% 30 minutes
+-define(ACCESS_CONTROL_ALLOW_CREDENTIALS , <<"true">>).
+
+%%%===================================================================
+%%% Behaviour API
+%%%===================================================================
+
+-spec execute(cowboy_req:req(), cowboy_middleware:env()) -> {ok, cowboy_req:req(), cowboy_middleware:env()}.
+execute(Req, Env) ->
+    Req2 = set_cors_headers(Req),
+    case cowboy_req:method(Req2) of
+        <<"OPTIONS">> -> {stop, cowboy_req:reply(200, Req2)};
+        _             -> {ok, Req2, Env}
+    end.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+-spec set_cors_headers(cowboy_req:req()) -> cowboy_req:req().
+set_cors_headers(Req1) ->
+    case cowboy_req:header(<<"origin">>, Req1) of
+        undefined ->
+            Req1;
+        Origin ->
+            case is_allowed(Origin, ?ACCESS_CONTROL_ALLOWED_DOMAINS) of
```

Comment:
🙂

Reply:
As I wrote, I just dropped the configuration layer, but kept the original structure for add it back.

Not perfect, I agree.